includes/admin/tools/class-settings-import.php

```diff
@@ -0,0 +1,681 @@
+            </tr>
+
+            <tr valign="top" class="give-import-dropdown">
+                <th><b><?php esc_html_e( 'Column name', 'give' ); ?></b></th>
+                <th><b><?php esc_html_e( 'Map to field', 'give' ); ?></b></th>
+            </tr>
+
+			<?php
+			$raw_key   = self::get_importer( $csv );
+			$donations = give_import_donations_options();
+			$donors    = give_import_donor_options();
+			$forms     = give_import_donation_form_options();
+
+			foreach ( $raw_key as $index => $value ) {
+				?>
+                <tr valign="top" class="give-import-option">
+                    <th><?php echo $value; ?></th>
+                    <th>
+						<?php
+						self::get_columns( $index, $donations, $donors, $forms, $value );
+						?>
+                    </th>
+                </tr>
+				<?php
+			}
+		}
+
+		static function selected( $option_value, $value ) {
+			$selected = '';
+			if ( stristr( $value, $option_value ) ) {
+				$selected = 'selected';
+			} elseif ( strrpos( $value, '_' ) && stristr( $option_value, 'Import as Meta' ) ) {
+				$selected = 'selected';
+			}
+
+			return $selected;
+		}
+
+		/**
+		 * Print the colums from the CSV.
+		 */
+		static function get_columns( $index, $donations, $donors, $forms, $value = false ) {
+			?>
+            <select name="mapto[<?php echo $index; ?>]">
+                <optgroup label="Donations">
+					<?php
+					foreach ( $donations as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Donors">
+					<?php
+					foreach ( $donors as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Forms">
+					<?php
+					foreach ( $forms as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+				<?php
+				do_action( 'give_import_dropdown_option', $index, $donations, $donors, $forms, $value );
+				?>
+            </select>
+			<?php
+		}
+
+		static function get_csv_total( $file_id ) {
+			$total = false;
+			if ( $file_id ) {
+				$file_dir = get_attached_file( $file_id );
+				if ( $file_dir ) {
+					$file = new SplFileObject( $file_dir, 'r' );
+					$file->seek( PHP_INT_MAX );
+					$total = $file->key() + 1;
+				}
+			}
+
+			return $total;
+		}
+
+		static function get_importer( $file_id, $index = 0, $delimiter = ',' ) {
```

Comment:
@raftaar1191 Please add documentation of function if newly created

Reply:
Done